mlos_bench/mlos_bench/optimizer/base_optimizer.py

```diff
@@ -23,56 +23,7 @@ class Optimizer(metaclass=ABCMeta):
     An abstract interface between the benchmarking framework and mlos_core optimizers.
     """
 
-    @staticmethod
-    def load(tunables: TunableGroups, config: dict, global_config: Optional[dict] = None) -> "Optimizer":
-        """
-        Instantiate the Optimizer shim from the configuration.
-
-        Parameters
-        ----------
-        tunables : TunableGroups
-            Tunable parameters of the environment.
-        config : dict
-            Configuration of the optimizer.
-        global_config : dict
-            Global configuration parameters (optional).
-
-        Returns
-        -------
-        opt : Optimizer
-            A new Optimizer instance.
-        """
-        (class_name, opt_config) = prepare_class_load(config, global_config)
-        opt = Optimizer.new(class_name, tunables, opt_config)
-        _LOG.info("Created optimizer: %s", opt)
-        return opt
-
-    @classmethod
-    def new(cls, class_name: str, tunables: TunableGroups, config: dict) -> "Optimizer":
-        """
-        Factory method for a new optimizer with a given config.
-
-        Parameters
-        ----------
-        class_name: str
-            FQN of a Python class to instantiate, e.g.,
-            "mlos_bench.optimizer.MlosCoreOptimizer".
-            Must be derived from the `Optimizer` class.
-        tunables : TunableGroups
-            The tunables to optimize.
-        config : dict
-            Free-format dictionary that contains the optimizer configuration.
-            It will be passed as a constructor parameter of the class
-            specified by `class_name`.
-
-        Returns
-        -------
-        opt : Optimizer
-            An instance of the `Optimizer` class initialized with `config`.
-        """
-        return instantiate_from_config(cls, class_name, tunables, config)
-
-    def __init__(self, tunables: TunableGroups, config: dict):
+    def __init__(self, tunables: TunableGroups, service: Optional[Service], config: dict):
```

Comment:
Total nit, but it'd be nice if we were consistent between the different class structures on order and/or required naming of the params.
I think for Environment, it's "config, global_config, tunables, service", for instance.
Here, there's an optional param that appears before the config.  It's just a little haphazard.
Maybe cleanup slighly in a future PR?

Reply:
agree. I just want to be consistent in which order we define these parameters across different classes. We can enforce that in abstract methods, but not in teh constructor, unforttunately